tools/server-spring/src/main/java/org/eclipse/rdf4j/http/server/repository/handler/AbstractQueryRequestHandler.java

```diff
@@ -114,17 +112,25 @@ public ModelAndView handleQueryRequest(HttpServletRequest request, RequestMethod
 			}
 
 		} catch (Exception e) {
-			// only close the connection when an exception occurs. Otherwise, the QueryResultView will take care of
-			// closing it.
-			if (repositoryCon != null) {
-				repositoryCon.close();
+			// only close the response & connection when an exception occurs. Otherwise, the QueryResultView will take
+			// care of closing it.
+			try {
+				if (queryResponse != null) {
```

Comment:
Maybe you can simplify your code down to an instanceOf AutoClosable check here without having to introduce the Boolean query stuff.